test/routes/api/user/user.test.ts

```diff
@@ -0,0 +1,179 @@
+
+async function createUser (
+  app: FastifyInstance,
+  userData: Partial<{ username: string; password: string }>
+) {
+  const [id] = await app.knex('users').insert(userData)
+  return id
+}
+
+async function deleteUser (
+  app: FastifyInstance,
+  userData: Partial<{ username: string; password: string }>
+) {
+  await app.knex('users').delete().where({ username: userData.username })
+}
+
+describe('User API', (t) => {
+  before(async () => {
+    const app = await build()
+    // Fill the password with the hashed value of `Password123$`
+    const Password123$ = 'ff57faf149a2bcab41bf7ecbbc8ce491.3ce6b34ea3edb3f0a09f811440885bfeda612832c04bfddc9d4b906019d97fa0'
+
+    await createUser(app, {
+      username: 'random-user-0',
+      password: Password123$
+    })
+
+    await createUser(app, {
+      username: 'random-user-1',
+      password: Password123$
+    })
+
+    await createUser(app, {
+      username: 'random-user-2',
+      password: Password123$
+    })
+
+    await createUser(app, {
+      username: 'random-user-3',
+      password: Password123$
+    })
+
+    await createUser(app, {
+      username: 'random-user-4',
+      password: Password123$
+    })
+
+    await app.close()
+  })
+
+  it('Should update the password successfully', async (t) => {
+    const app = await build(t)
+
+    const res = await app.injectWithLogin('random-user-0', {
+      method: 'PUT',
+      url: '/api/user/update-password',
+      payload: {
+        currentPassword: 'Password123$',
+        newPassword: 'NewPassword123$'
+      }
+    })
+
+    assert.strictEqual(res.statusCode, 200)
+    assert.deepStrictEqual(JSON.parse(res.payload), { message: 'Password updated successfully' })
+
+    await deleteUser(app, {
+      username: 'random-user-0'
+    })
+  })
+
+  it('Should return 400 if the new password is the same as current password', async (t) => {
+    const app = await build(t)
+
+    const res = await app.injectWithLogin('random-user-1', {
+      method: 'PUT',
+      url: '/api/user/update-password',
+      payload: {
+        currentPassword: 'Password123$',
+        newPassword: 'Password123$'
+      }
+    })
+
+    assert.strictEqual(res.statusCode, 400)
+    assert.deepStrictEqual(JSON.parse(res.payload), { message: 'New password cannot be the same as the current password.' })
+
+    await deleteUser(app, {
+      username: 'random-user-1'
+    })
+  })
+
+  it('Should return 400 if the newPassword password not match the required pattern', async (t) => {
+    const app = await build(t)
+
+    const res = await app.injectWithLogin('random-user-2', {
+      method: 'PUT',
+      url: '/api/user/update-password',
+      payload: {
+        currentPassword: 'Password123$',
+        newPassword: 'password123$'
+      }
+    })
+
+    assert.strictEqual(res.statusCode, 400)
+    assert.deepStrictEqual(JSON.parse(res.payload), { message: 'body/newPassword must match pattern "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).*$"' })
```

Comment:
I should find a way to return a more explicit error message

Reply:
> Password must contain at least one uppercase letter, one lowercase letter, one digit, and one special character (#?!@$%^&*-).